src/axom/slic/interface/slic_macros.hpp

```diff
@@ -70,6 +71,10 @@
       __oss << msg;                                                 \
       axom::slic::logErrorMessage(__oss.str(), __FILE__, __LINE__); \
     }                                                               \
+    if(axom::slic::isInitialized())                                 \
+    {                                                               \
+      axom::slic::abortIfEnabled(axom::slic::message::Error);       \
+    }                                                               \
```

Comment:
Two related points:
- Shouldn't the call to `abortIfEnabled` only happen if `EXP` is true?  So should go inside the close-bracket at (new) line73?
- `slic::logErrorMessage()` eventually calls `ensureInitialized()` at slic.cpp (new) line 221, so we should not need to test on the result of `isInitialized()`.

Reply:
- `abortIfEnabled` is how we collectively check that a rank has aborted. If we put  `abortIfEnabled()` inside the `EXP` block, then every rank must have `EXP` evaluate to the same value to call `abortIfEnabled`  or MPI will likely hang the application. The name `"abortIfEnabled"` is potentially not so clear, and does not stress that it must to be called collectively... Open to suggestions for names to all of the added "*abort*"  functions.
- Without the  `isInitialized()` block,` ensureInitialized() `will always be called from  `abortIfEnabled()`, even when no message has been logged by SLIC. This leads to `ensureInitialized()` warning messages for almost  every unit test that eventually hits a `SLIC_ASSERT`.

Reply:
I think the naming of `abortIfEnabled` was part of what confused me, though it should not have.  Maybe consider one of 
- abortIfSignaled (hmmm, might connote operating system signals)
- abortIfFlagged
- abortIfRequested (like the stop request pull cord on a bus?)
- abortIfSomeoneRequested (eh, that's getting silly)

Thank you for explaining both points.

Reply:
`abortIfAbortOnWarningOrErrorIsEnabled()`?

Reply:
Hmmm....  thinking on it more, `abortIfEnabled(level)` fits well with the `enableAbortOnXXX` and `disableAbortOnXXX` functions. I like the verbs `enable/disable` for these functions (I can't think of a better replacement), and it makes sense for `abortIfXXX` to use the same verbiage. Kind of convinced myself to keep it the same.

Reply:
My thinking is more inline with @agcapps.

How about something like this:

```cpp
  do
  {
    if(EXP)
    {
      std::ostringstream __oss;
      __oss << msg;
      // Note: the following internally calls Logger::setAbortFlag() when necessary
      // using Logger's m_abortOnError / m_abortOnWarning flags
      axom::slic::logErrorMessage(__oss.str(), __FILE__, __LINE__);       
    }
    // Note: Abort will only be raised locally/globally when m_abortOnError and/or m_abortOnWarning
    // So, the following will only need to be collective when m_abortOnError
    // and the underlying LogStream is MPI-based
    if(axom::slic::checkAbortRaised(message::Level))
    { 
      axom::slic::flushStreams();  // note: all ranks must already participate for MPI-based LogStreams
      axom::slic::abort();   // note: no message::Level
    }
  } while(axom::slic::detail::false_value)
```

Reply:
@bmhan12 wrote:
> Without the isInitialized() block,ensureInitialized()will always be called from abortIfEnabled(), even when no message has been logged by SLIC. This leads to ensureInitialized() warning messages for almost every unit test that eventually hits a SLIC_ASSERT, with nothing actually logged.

We should not have any unit tests that use slic without initializing it first.
I'd argue that this is a feature, and we should fix any test files with this problem.